ocaml/xapi/repository_helpers.ml

```diff
@@ -51,47 +123,95 @@ module Pkg = struct
     | false ->
         None
 
-  let to_ver_rel_json pkg =
-    `Assoc [("version", `String pkg.version); ("release", `String pkg.release)]
+  let to_epoch_ver_rel_json pkg =
+    `Assoc
+      [
+        ("epoch", `String (Epoch.to_string pkg.epoch))
+      ; ("version", `String pkg.version)
+      ; ("release", `String pkg.release)
+      ]
 
   let to_name_arch_string pkg = pkg.name ^ "." ^ pkg.arch
+
+  let to_fullname ~name ~arch ~epoch ~version ~release =
+    match epoch with
+    | Some i ->
+        Printf.sprintf "%s-%s:%s-%s.%s.rpm" name (string_of_int i) version
+          release arch
+    | None ->
+        Printf.sprintf "%s-%s-%s.%s.rpm" name version release arch
 end
 
 module Update = struct
   type t = {
       name: string
     ; arch: string
+    ; old_epoch: Epoch.t option
```

Comment:
Could you explain why is `Epoch.t option` used here instead of `Epoch.t`? I couldn't spot a difference in behaviour between `None` and `Some None`. This change would also simplify the code around comparisons.

Reply:
An update has:
1. new package, this comes from `yum list updates`
2. old package, this comes from `rpm -qa`

Since they come from different sources, one maybe case is that a package in `yum list updates` but not in `rpm -qa`. 
And meanwhile, an epoch itself might be None as well. So epoch is represented as `Epoch.t`(`int option`), while the old epoch is  represented as `Epoch.t option`